src/core/load-fragment.c

```diff
@@ -188,6 +263,16 @@ int config_parse_unit_deps(
                         continue;
                 }
 
+                r = unit_is_recursive_template_dependency(u, k, word);
+                if (r < 0)
+                        return r;
+                if (r > 0) {
+                        log_syntax(unit, LOG_WARNING, filename, line, r,
```

Comment:
if this is a valid use case, then this shouldn't be a warning - better to document the dropping behaviour, and make this a debug or info at most